LEGO1/lego/legoomni/src/worlds/hospital.cpp

```diff
@@ -559,16 +559,16 @@ MxBool Hospital::HandleControl(LegoControlManagerNotificationParam& p_param)
 			BackgroundAudioManager()->RaiseVolume();
 			DeleteObjects(&m_atomId, HospitalScript::c_hho002cl_RunAnim, HospitalScript::c_hho006cl_RunAnim);
 
-			if (m_unk0x100 == 1) {
-				m_hospitalState->m_unk0x08 = 14;
+			if (m_interactionMode == 1) {
+				m_hospitalState->m_state = HospitalState::e_exitViaInfo;
```

Comment:
```suggestion
				m_hospitalState->m_state = HospitalState::e_exitViaDoor;
```
I've noticed that clicking the info/door results in different destinations depending on when you click them. Should I fix it here or in a different pull request?

If this change should only convert numbers into their constants it should be changed here. If the issue should be fixed, I would need to change the state below.

I've compared it to the original game and there it works as expected and not depending on the timing.

Reply:
It seems that the correct value should be exitViaInfo, since it maps to 14.

Below, it should be exitViaDoor because it's 15.

We should leave the mappings the way they are - according to the accuracy comparison, HandleControl matches 100% so functionally it should be correct and act the same way as the original. If you see a difference in game behavior regardless (please double check), the issue is likely not to be found in this function.

https://github.com/isledecomp/isle/actions/runs/15517986153/job/43687412923?pr=1545 You can see that currently based on the state of the branch the accuracy has regressed. This is likely to the use of exitViaInfo

Reply:
Okay I'm going to double check this.

It seems to me that you switched up the values for `exitVia...` and that this change should be applied to `exitViaDoor` (at least according to the values they represent). I named them this way, because they lead to the "normal" behavior (`exitViaDoor` positions you outside the building and `exitViaInfo` spawns you in the info center). Should I name them the over way around?

Reply:
Sorry, I'm a bit confused with the naming changes. What matters in the end that the values don't change. I'll leave the naming up to you.